### PR TITLE
Yeah this version ius wayyy more better

### PR DESCRIPTION
**Pull request by FoxWorn3365**

### CODE REVIEW

**Comment by SpGerg** [comment]
again this italians...

File: UncomplicatedCustomItems/API/Events.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            EventList[Event] = new();
+        }
+
+        public static void Unregister()
+        {
+            EventList.Clear();
+        }
+
+        internal static bool _Trigger(ItemEvents Event, CustomItemHandler Handler)
```
Comment (SpGerg) [inline UncomplicatedCustomItems/API/Events.cs:44]: _Trigger?

File: UncomplicatedCustomItems/API/Features/CustomItemHandler.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        public Dictionary<ItemEvents, Func<CustomItemHandler, bool>> InternalEvents { get; }
+
+        public CustomItemHandler(ICustomItem customItem, Player player)
+        {
+            CustomItem = customItem;
+            Player = player;
+            Item = Item.Create(customItem.Item, player);
+            Serial = Item.Serial;
+            InternalEvents = new();
+        }
```
Comment (SpGerg) [inline UncomplicatedCustomItems/API/Features/CustomItemHandler.cs:28]: the constructor must be above the fields

File: UncomplicatedCustomItems/API/Features/CustomItemHandler.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        public void RemoveEventListener(ItemEvents Event)
+        {
+            InternalEvents.Remove(Event);
+        }
+
+        internal bool TriggerEvent(ItemEvents Event)
+        {
+            bool Allowed = true;
```
Comment (SpGerg) [inline UncomplicatedCustomItems/API/Features/CustomItemHandler.cs:42]: var

File: UncomplicatedCustomItems/API/Features/CustomItemHandler.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            }
+
+            if (CustomItem.Event == Event)
+            {
+                if (CustomItem.Command is not null && CustomItem.Command != string.Empty)
+                {
+                    Server.ExecuteCommand(CustomItem.Command.Replace("%id%", Player.Id.ToString()), Player.Sender);
+                }
+
+                Helper.Helper.ParseResponse(Player, CustomItem.Response);
```
Comment (SpGerg) [inline UncomplicatedCustomItems/API/Features/CustomItemHandler.cs:59]: Typically such classes are called "Utils" or "Utilities".

File: UncomplicatedCustomItems/Commands/UCIUse.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+            response = "";
+            Player Player = Player.Get(sender);
+            if (Helper.Helper.IsCustomItem(Player, Player.CurrentItem))
+            {
+                Helper.Helper.GetCustomItem(Player, Player.CurrentItem).TriggerEvent(ItemEvents.Command);
+            }
+            return true;
+        }
+    }
+}
```
Comment (SpGerg) [inline UncomplicatedCustomItems/Commands/UCIUse.cs:31]: Delete it, im will make own command system.

File: UncomplicatedCustomItems/Events/Internal/Player.cs [hunk 6]
```diff
@@ -1,33 +1,25 @@
...
+            if (!Helper.Helper.HandleItemEvent(ItemEvents.Drop, PickingUp, PickingUp.Player))
             {
-                return;
+                PickingUp.Player.DropItem(PickingUp.Item);
             }
-
-            ev.IsAllowed = !ev.Item.IsCustomItem();
         }
     }
 }
```
Comment (SpGerg) [inline UncomplicatedCustomItems/Events/Internal/Player.cs:25]: Delete the your changes

File: UncomplicatedCustomItems/Helper/Helper.cs [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+            {
+                Player.Broadcast(Response.BroadcastDuration, Response.BroadcastMessage);
+            }
+
+            if (Response.HintMessage.Length > 1 && Response.HintDuration > 0)
+            {
+                Player.ShowHint(Response.HintMessage, Response.HintDuration);
+            }
+        }
+    }
```
Comment (SpGerg) [inline UncomplicatedCustomItems/Helper/Helper.cs:159]: Docs?

File: UncomplicatedCustomItems/Interfaces/ICustomItem.cs [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+        public abstract Vector3 Scale { get; set; }
+
+        public abstract ItemEvents Event { get; set; }
+        
+        public abstract string? Command { get; set; }
+
+        public abstract Response? Response { get; set; }
+
+        public abstract bool EventsEnabled { get; set; }
```
Comment (SpGerg) [inline UncomplicatedCustomItems/Interfaces/ICustomItem.cs:25]: Docs

File: UncomplicatedCustomItems/Interfaces/IResponse.cs [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+        public abstract string ConsoleMessage { get; set; }
+
+        public abstract string BroadcastMessage { get; set; }
+
+        public abstract ushort BroadcastDuration { get; set; }
+
+        public abstract string HintMessage { get; set; }
+
+        public abstract float HintDuration { get; set; }
+    }
```
Comment (SpGerg) [inline UncomplicatedCustomItems/Interfaces/IResponse.cs:14]: Docs

File: UncomplicatedCustomItems/Plugin.cs [hunk 10]
```diff
@@ -1,28 +1,51 @@
...
+        public override Version RequiredExiledVersion => new(8, 8, 1);
 
-        public static UncomplicatedCustomItemsAPI API { get; private set; }
+        // Custom var for plugin management
+        public static Plugin Instance { get; private set; }
+        internal EventHandler Handler;
+        // Item Id => Item
+        internal static Dictionary<uint, ICustomItem> Items { get; private set; } = new();
+        // Player ID => List<>() of custom items
+        internal static Dictionary<int, List<CustomItemHandler>> ItemDictionary { get; private set; } = new();
```
Comment (SpGerg) [inline UncomplicatedCustomItems/Plugin.cs:27]: Line between fields. Also, why you remove the Dictionary from api class? In my opinion, it is better not to store all the values ​​in the main class

File: UncomplicatedCustomItems/Plugin.cs [hunk 11]
```diff
@@ -1,28 +1,51 @@
...
 
         public override void OnDisabled()
         {
-            Events.Internal.Player.Unregister();
+            EventSource.DroppingItem -= Handler.OnItemDropping;
+            EventSource.ItemAdded -= Handler.OnItemPickup;
+            EventSource.UsingItem -= Handler.OnItemUsing;
 
+            Handler = null;
+            Instance = null;
```
Comment (SpGerg) [inline UncomplicatedCustomItems/Plugin.cs:48]: it's not obligatory

**Review by SpGerg — CHANGES_REQUESTED** [review]

File: UncomplicatedCustomItems/API/Features/CustomItemHandler.cs (hunk 3 above)
Comment (FoxWorn3365) [reply UncomplicatedCustomItems/API/Features/CustomItemHandler.cs:42]: Does not make sense, is better to use bool

File: UncomplicatedCustomItems/API/Features/CustomItemHandler.cs (hunk 4 above)
Comment (FoxWorn3365) [reply UncomplicatedCustomItems/API/Features/CustomItemHandler.cs:59]: Tipically

File: UncomplicatedCustomItems/Commands/UCIUse.cs (hunk 5 above)
Comment (FoxWorn3365) [reply UncomplicatedCustomItems/Commands/UCIUse.cs:31]: Just do it

File: UncomplicatedCustomItems/Events/Internal/Player.cs (hunk 6 above)
Comment (FoxWorn3365) [reply UncomplicatedCustomItems/Events/Internal/Player.cs:25]: Why?

File: UncomplicatedCustomItems/Helper/Helper.cs (hunk 7 above)
Comment (FoxWorn3365) [reply UncomplicatedCustomItems/Helper/Helper.cs:159]: You made me rush

File: UncomplicatedCustomItems/Interfaces/ICustomItem.cs (hunk 8 above)
Comment (FoxWorn3365) [reply UncomplicatedCustomItems/Interfaces/ICustomItem.cs:25]: The name explain itself

File: UncomplicatedCustomItems/Interfaces/IResponse.cs (hunk 9 above)
Comment (FoxWorn3365) [reply UncomplicatedCustomItems/Interfaces/IResponse.cs:14]: Not needed?

File: UncomplicatedCustomItems/Plugin.cs (hunk 10 above)
Comment (FoxWorn3365) [reply UncomplicatedCustomItems/Plugin.cs:27]: The safety is increased

File: UncomplicatedCustomItems/Plugin.cs (hunk 11 above)
Comment (FoxWorn3365) [reply UncomplicatedCustomItems/Plugin.cs:48]: Obbligatory != reccomended

**Comment by FoxWorn3365** [comment]
another is arriving, let's close that

**Pull request closed without merging**